src/mod/codecs/mod_amrwb/mod_amrwb.c

```diff
@@ -401,13 +406,15 @@ static switch_status_t switch_amrwb_decode(switch_codec_t *codec,
 	return SWITCH_STATUS_FALSE;
 #else
 	struct amrwb_context *context = codec->private_info;
-	unsigned char *buf = encoded_data;
+	unsigned char buf[SWITCH_AMRWB_OUT_MAX_SIZE];
 	uint8_t tmp[SWITCH_AMRWB_OUT_MAX_SIZE];
 
 	if (!context) {
 		return SWITCH_STATUS_FALSE;
 	}
 
+	memcpy(buf, encoded_data, encoded_data_len);
```

Comment:
Potential stack overflow if `encoded_data_len > SWITCH_AMRWB_OUT_MAX_SIZE` by any reason.

Reply:
As input frames are never supposed to be larger than SWITCH_AMRWB_OUT_MAX_SIZE and it is really an error case, would you be okay with returning SWITCH_STATUS_FALSE and not processing such frames?

Reply:
Yeah. Let's drop it.

Reply:
Added extra checks of encoded_data_len

Reply:
@andywolk @dkunilov I've detected a case where the `encoded_data_len > SWITCH_AMRWB_OUT_MAX_SIZE` comparison causes calls to be hung up during a re-INVITE where the other party responds with multiple codecs in their 200 OK.

Alice and Bob are in a call using PCMA, and FreeSWITCH sends a re-INVITE to Alice. Alice responds with all codecs she accepts, with AMR-WB at the top. FreeSWITCH momentarily assumes the call is switching to AMR-WB, but Alice continues to send PCMA. With the changes from this PR, FreeSWITCH hangs up the call after `switch_amrwb_decode` returns `SWITCH_STATUS_FALSE`. Here's a log excerpt:

> 2024-04-23 13:12:44.593843 91.27% [DEBUG] sofia.c:8524 Processing updated SDP
2024-04-23 13:12:44.593843 91.27% [DEBUG] switch_core_media.c:8654 Audio params are unchanged for sofia/foo/Alice@X.X.X.X.
2024-04-23 13:12:44.593843 91.27% [DEBUG] switch_core_media.c:3644 Changing Codec from PCMA@20ms@8000hz to AMR-WB@20ms@16000hz
2024-04-23 13:12:44.613817 91.27% [DEBUG] switch_rtp.c:4437 RE-Starting timer [soft] 320 bytes per 20ms
2024-04-23 13:12:44.613817 91.27% [DEBUG] switch_core_media.c:3743 Set Codec sofia/foo/Alice@X.X.X.X AMR-WB/16000 20 ms 320 samples 23850 bits 1 channels
2024-04-23 13:12:44.613817 91.27% [DEBUG] switch_core_codec.c:124 sofia/foo/Alice@X.X.X.X Original read codec replaced with AMR-WB:110
2024-04-23 13:12:44.654008 91.27% [DEBUG] switch_core_media.c:3199 alternate payload received (received 8, expecting 96)
2024-04-23 13:12:44.654008 91.27% [WARNING] switch_core_media.c:3210 Changing current codec to PCMA (payload type 8).
2024-04-23 13:12:44.654008 91.27% [ERR] switch_core_media.c:16066 Codec AMR-WB / Bandwidth Efficient decoder error!
2024-04-23 13:12:44.654008 91.27% [DEBUG] switch_ivr_bridge.c:819 sofia/foo/Bob ending bridge by request from write function

I have not yet deduced what the encoded_data actually contains, if it actually inserts the PCMA payload into the AMR-WB decoder or what's going on, but I have tested the same case without mod_amrwb loaded (so it momentarily chooses g722) and that works perfectly.

Reply:
A description of how I encountered this:

1. Incoming call from Alice to FreeSWITCH, with Alice offering AMR-WB, G722, AMR, PCMA, PCMU.
2. Alice is bridged to Bob, using late negotiation.
3. Bob answers with PCMA.
4. FreeSWITCH connects Bob with Alice, both using PCMA.
5. FreeSWITCH sends re-INVITEs to both Alice and Bob, containing only PCMA.
6. Alice detects no change in SDP, so she answers with her previous offer (the initial offer, containing _all_ codecs).
7. FreeSWITCH seemingly tries to switch to AMR-WB, and the call breaks.

Alice's behaviour in step 6 is a bit puzzling to me, but I see nothing in any RFC which says it is incorrect.

Reply:
@andywolk @dkunilov I found out that the above re-INVITE sequence actually resulted in a PCMA payload being inserted into `encoded_data`. A fix for this is available in #2452.